src/lib/support/logging/TextOnlyLogging.h

```diff
@@ -98,6 +98,27 @@ using LogRedirectCallback_t = void (*)(const char * module, uint8_t category, co
 #define ChipLogError(MOD, MSG, ...) ((void) 0)
 #endif // CHIP_ERROR_LOGGING
 
+/**
+ * @def ChipLogFailure(err, MOD, MSG, ...)
+ *
+ * @brief
+ *   Log a chip message with a formatted ChipError for the specified module in the 'Error'
+ *   category.
+ *
+ *  Example usage:
+ *
+ *  @code
+ *    ChipLogFailure(errCode, Module, "Failure message: %s", param);
+ *  @endcode
+ *
+ *  @param[in]  ERROR_CODE  A CHIP_ERROR to log.
+ *  @param[in]  MOD         The log module to use.
+ *  @param[in]  MSG         The log message format string.
+ *  @param[in]  ...         Optional arguments for the log message.
+ */
+#define ChipLogFailure(ERROR_CODE, MOD, MSG, ...)                                                                                  \
```

Comment:
Approving, however it is a bit annoying that a `ChipLog...` macro does not start with the module (like all the others do). Maybe `ChipLogFailure(MOD, ERR, MSG, ...) would be easier to read. :shrug:

Reply:
Yes, maybe we do not fix everything, just make life better.

I am still sad that we reverted the name to `CodeUtils` when  something like `ErrorReportingMacros.h` would have been a vastly better name...